### ISSUE TITLE
The patched SVC model, mostly has different results from unpatched SVC( the raw sklearn version)

### ISSUE THREAD

**Issue by 11Sadegh11** [intel/scikit-learn-intelex#1074]
<h3>Description:</h3>

I tried different imbalance dataset from imblearn.dataset (i.e. namely sick_euthyroid, wine_quality, optical_digits, satimage and mammography)  by patched version of SVC(i.e. with patched_sklearn() ) and normal sklearn SVC version(i.e. unpatched SVC), all with 10 random different **C and gamma** hyperparameter. Surprisingly, I see that for most of them, the **F1 scores** of the patched and unpatched SVC are totally different !!
<h3>To Reproduce:</h3>

**To get each data set:** (followings is an example) 
`from imblearn.datasets import fetch_datasets`
`from sklearn.preprocessing import StandardScaler`
`wine                 = fetch_datasets()['wine_quality']`
`X_wine, Y_wine = wine.data, wine.target`
`X_wine_norm    = StandardScaler().fit_transform(X=X_wine)`
**To make cross validation process:**
`from sklearn.model_selection import RepeatedStratifiedKFold, cross_val_score`
`K_Fold     = RepeatedStratifiedKFold(n_splits=5,n_repeats=2, random_state= 42)`
`score_obj = cross_val_score(estimator= clf, X=X, y=Y, scoring= 'f1', cv= K_Fold).mean()`
<h3> Expected behavior:</h3>

To get about the same **F1 score** for all or most pairs of the Patched and Unpatched SVC objects.
<h3> The Results:</h3>

To prevent this post from becoming crowded, I have attached the results as a .txt file.
**The Attachments:**
[Patched vs Unpatched SVC.txt](https://github.com/intel/scikit-learn-intelex/files/10148946/Patched.vs.Unpatched.SVC.txt)

<h3>Note:</h3>

It worth to mention that also, I tried SVC from **daal4py.sklearn.svm** and the **F1 scores** of this model was totally **the same** as the **normal** sklearn SVC model.(but the runing time of **daal4py** version was a little more than that with **patched_sklearn()** SVC model)

**Comment by Alexsandruss** [comment]
`sklearnex` can produce results which are bit-to-bit similar to default sklearn only for subset of algorithms and parameters.
I'm reproducing this issue with `optuna` hyperparameters search framework, but only for `sigmoid` svm kernel. `poly` and `rbf` kernels produce same f1 scores
Could you specify all your svm parameters, please?

**Comment by 11Sadegh11** [comment]
> `sklearnex` can produce results which are bit-to-bit similar to default sklearn only for subset of algorithms and parameters. I'm reproducing this issue with `optuna` hyperparameters search framework, but only for `sigmoid` svm kernel. `poly` and `rbf` kernels produce same f1 scores Could you specify all your svm parameters, please?

Sure, Actually I was running hyperparameter tuning by Optuna that the 10 first hyperparameters which were totally the same( as a result of setting the same random_state), have mostly different F1 scores which I reported them here.
By the way, the followings is the code I have used:
```
################################ Optuna Objective Function ####################################
##### Objective-Function ((Patched)) SVC #####
patch_sklearn()
from sklearn.svm import SVC
def objective_svc(trial, X, Y):
    K_Fold    = RepeatedStratifiedKFold(n_splits=5,n_repeats=2, random_state= 42)
    C_par     = trial.suggest_float('C_par', 1e-3, 1e+3, log=True)
    gamma_par = trial.suggest_float('gamma_par', 1e-3, 1e+3, log=True)
    clf       = SVC(C= C_par, gamma= gamma_par, class_weight= 'balanced')
    score_obj = cross_val_score(estimator= clf, X=X, y=Y, scoring= 'f1', cv= K_Fold).mean()
    return score_obj

start= time.time()
model_study = optuna.create_study(direction= 'maximize', sampler= optuna.samplers.TPESampler(seed= 42))
model_study.optimize(lambda trial: objective_svc(trial, X, y), n_trials= 50)
C_best, gamma_best = model_study.best_params['C_par'], model_study.best_params['gamma_par'] ;
clf_best_Patch     = SVC(C= C_best, gamma= gamma_best, class_weight= 'balanced') ;
print(f"The Hypars Tuning's Runing Time of Patched SVC is = {time.time() - start}")

##### Objective-Function ((Unpatched)) SVC #####

unpatch_sklearn()
from sklearn.svm import SVC
def objective_svc(trial, X, Y):
    K_Fold    = RepeatedStratifiedKFold(n_splits=5,n_repeats=2, random_state= 42)
    C_par     = trial.suggest_float('C_par', 1e-3, 1e+3, log=True)
    gamma_par = trial.suggest_float('gamma_par', 1e-3, 1e+3, log=True)
    clf       = SVC(C= C_par, gamma= gamma_par, class_weight= 'balanced')
    score_obj = cross_val_score(estimator= clf, X=X, y=Y, scoring= 'f1', cv= K_Fold).mean()
    return score_obj

start = time.time()
model_study = optuna.create_study(direction= 'maximize', sampler= optuna.samplers.TPESampler(seed= 42))
model_study.optimize(lambda trial: objective_svc(trial, X, y), n_trials= 50)
C_best, gamma_best = model_study.best_params['C_par'], model_study.best_params['gamma_par'] ;
clf_best_Unptch    = SVC(C= C_best, gamma= gamma_best, class_weight= 'balanced') ;
print(f"The Hypars Tuning's Runing Time of Unpatched SVC is = {time.time() - start}")
```

**Comment by Alexsandruss** [comment]
Reason of issue is incorrect handling of `class_weight='balanced'` in sklearnex. I will create PR to fix it

**Comment by 11Sadegh11** [comment]
> Reason of issue is incorrect handling of `class_weight='balanced'` in sklearnex. I will create PR to fix it

That would be really great.
Thanks for your fast responding and caring about that.

**Comment by Alexsandruss** [comment]
Fixed in #1080. However, sklearnex and default sklearn may produce different predictions for specific datasets because of different implementations of SVM, but difference is not dramatic now with fix.
